mongo/dataset_store.go

```diff
@@ -271,14 +272,21 @@ func (m *Mongo) UpdateDataset(id string, dataset *models.Dataset) (err error) {
 	s := m.Session.Copy()
 	defer s.Close()
 
-	updates := createDatasetUpdateQuery(dataset)
-	err = s.DB(m.Database).C("datasets").UpdateId(id, bson.M{"$set": updates, "$setOnInsert": bson.M{"next.last_updated": time.Now()}})
+	updates := createDatasetUpdateQuery(id, dataset)
+	if err = s.DB(m.Database).C("datasets").UpdateId(id, bson.M{"$set": updates, "$setOnInsert": bson.M{"next.last_updated": time.Now()}}); err != nil {
+		if err == mgo.ErrNotFound {
+			return errs.DatasetNotFound
+		}
```

Comment:
This handles the 'ErrNotFound' but would swallow any other error? Consider handling other errors here.

Reply:
Any other error will still be returned and handled in the api package (most likely to result in a 500 internal server error)

Reply:
Ill change it so it is explicit